src/radio-button/radio-button.styles.tsx

```diff
@@ -22,15 +22,23 @@ export const Container = styled.div<StyleProps>`
     width: 1.5rem;
     position: relative;
     border-radius: 50%;
-    border: 2px solid ${V2_Color.Accent.Light[2]};
+    border: 2px solid ${Colour["icon-primary-subtlest"]};
 
     ${(props) => {
         if (props.$disabled) {
             return `
-				border: 2px solid ${V2_Color.Neutral[4](props)};
+				border: 2px solid ${Colour["border-disabled"](props)};
 			`;
         }
     }}
+
+    &:hover {
+        border: 2px solid
+            ${(props) =>
+                props.$disabled
+                    ? Colour["border-disabled"](props)
+                    : Colour["icon-hover"](props)};
+    }
```

Comment:
can consider a similar structure to Toggle to apply the color if:

- disabled
- not disabled
   - selected
   - default